addons/base_automation/models/ir_actions_server.py

```diff
@@ -11,3 +11,10 @@ class ServerAction(models.Model):
         ('base_automation', 'Automation Rule')
     ], ondelete={'base_automation': 'cascade'})
     base_automation_id = fields.Many2one('base.automation', string='Automation Rule', ondelete='cascade')
+
+    @api.depends('usage')
+    def _compute_available_model_ids(self):
+        rule_based = self.filtered(lambda action: action.usage == 'base_automation')
```

Comment:
Mmmmh.

You have one check based on "state", and one check based on "usage".
I don't really know how base.automation works, but it feels that if those are not mutually exclusive, then your computes could override each other and set "available_model_ids" based on which is done first, which seems wrong?

Maybe I'm just confused.

Reply:
It is a compute method: it will be computed anytime one of state and usage changes. And sms then mail will be executed first.